### DISCUSSION TITLE
How to manage keyboard focus & navigation with react-aria-components table?

### DISCUSSION THREAD

**Discussion by nirtamir2** [adobe/react-spectrum#5006] [Q&A] [answered]
I created a table grid with `@tanstack/table`. I want to add it to the keyboard navigation so I use the `react-aria-components` table.
I don't understand how to manage the focus/keyboard navigation. I get some good behavior by default. I want when the user navigates with the arrow keys - the focused & focusVisible cells will change from div to inputs - so the user will be able to edit them. If I create a non-focusable element, I notice that it will make the td element (Cell) focusable and will change the Cell render props `isFocus` and `isFocusVisible` to `true`. But if I have an element that has focus inside (like `input` / `tabIndex=0`), those render props won't be `true`, and the focus will be on this element. So it makes stuff hard to control in this way. Is there a way to control the focus management for the keyboard? Am I missing something with the implementation? My previous solution to this was to save the state for `selectedCell` and I render stuff according to this value. I wrap my cells with a component that listens to focus and emits `setSelectedCell` when it's called. Thanks!

**Comment by reidbarber** [comment, marked as answer]
In order to support an input inside a cell, we need to support edit mode for grids navigation. There is an open issue for this here: https://github.com/adobe/react-spectrum/issues/2328

**Reply by nirtamir2** [reply]
Thanks!
